internal/bson/array.go

```diff
@@ -29,9 +29,13 @@ type arrayType types.Array
 func (a *arrayType) bsontype() {}
 
 // ReadFrom implements bsontype interface.
-func (a *arrayType) ReadFrom(r *bufio.Reader) error {
+func (a *arrayType) ReadFrom(r *bufio.Reader, nesting int) error {
+	if nesting > maxNesting {
+		return lazyerrors.Errorf("bson.Document.ReadFrom (over nested document. Max supported nesting: %d)", maxNesting)
```

Comment:
Let's maybe also print `nesting` too?
Also, the message might be confusing for someone who sees it for the first time.
Maybe something like "the document has the depth X, max supported depth is Y"?

Also, this is not `bson.Document` but rather `bson.Array` I guess.

Reply:
As we don't proceed the document after exceeding the maxDepth, we don't know the actual depth of provided exceeding document